pkg/didcomm/protocol/didexchange/service_test.go

```diff
@@ -121,21 +121,9 @@ func TestService_Handle_Inviter(t *testing.T) {
 	statusCh := make(chan service.StateMsg, 10)
 	err = s.RegisterMsgEvent(statusCh)
 	require.NoError(t, err)
-	completeFlag := make(chan bool)
-	respondedFlag := make(chan bool)
-	go func() {
-		for e := range statusCh {
-			if e.Type == service.PostState {
-				// receive the events
-				if e.StateID == "completed" {
-					completeFlag <- true
-				}
-				if e.StateID == "responded" {
-					respondedFlag <- true
-				}
-			}
-		}
-	}()
+	completedFlag := make(chan struct{})
+	respondedFlag := make(chan struct{})
+	msgEventListener(t, statusCh, respondedFlag, completedFlag)
```

Comment:
personally i like doing go func name rather than embedding the go. it makes it more obvious that there is a goroutine.

Reply:
Linter complained about the complexity (> 10). Hence moved to new function.

Reply:
Misread the point. [Moved](https://github.com/hyperledger/aries-framework-go/compare/ebaa2c60d8b2fce620d906c585bfb49c74ba8111..cc46c9e091f6091fb11e5c1467352ad67d28c0ea#diff-36ac39bebf32030fd7dff6f12aea8d7eR126) go routine from function to caller.